src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,129 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+
+/**
+ * The Parser class takes in the input string and performs the operations
+ * based on the command found in the string.
+ */
+public class Parser {
+    private Ui ui;
+    private Storage storage;
+    private TaskList taskList;
+
+    /**
+     * Constructor for Parser.
+     *
+     * @param ui Ui instance for output.
+     * @param storage Storage instance for saving task.
+     * @param taskList TaskList instance for operations on tasks.
+     */
+    public Parser(Ui ui, Storage storage, TaskList taskList) {
+        this.ui = ui;
+        this.storage = storage;
+        this.taskList = taskList;
+    }
+
+    /**
+     * Processes the input string and decide what operations to perform.
+     *
+     * @param input The input String.
+     * @return Boolean value indicating if program should continue.
+     */
+    public boolean process(String input) {
```

Comment:
Should the different cases be extracted out as separate methods?
https://nus-cs2103-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-long-methods

> Be wary when a method is longer than the computer screen, and take corrective action when it goes beyond 30 LOC (lines of code). The bigger the haystack, the harder it is to find a needle.